src/containerapp/azext_containerapp/custom.py

```diff
@@ -1510,6 +1519,963 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
+        return update_containerapp_yaml(cmd=cmd, name=name, resource_group_name=resource_group_name, file_name=yaml, no_wait=no_wait)
+
+    containerappsjob_def = None
+    try:
+        containerappsjob_def = ContainerAppsJobClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
+    except:
+        pass
+
+    if not containerappsjob_def:
+        raise ResourceNotFoundError("The containerapps job '{}' does not exist".format(name))
+
+    new_containerappsjob = {}
+    new_containerappsjob["properties"] = {}
+
+    # Doing this while API has bug. If env var is an empty string, API doesn't return "value" even though the "value" should be an empty string
+    if "properties" in containerappsjob_def and "template" in containerappsjob_def["properties"] and "containers" in containerappsjob_def["properties"]["template"]:
+        for container in containerappsjob_def["properties"]["template"]["containers"]:
+            if "env" in container:
+                for e in container["env"]:
+                    if "value" not in e:
+                        e["value"] = ""
+
+    update_map = {}
+    update_map['replicaConfigurations'] = replica_timeout or replica_retry_limit
+    update_map['triggerConfigurations'] = replica_completion_count or parallelism or cron_expression or scale_rule_name
+    update_map['container'] = image or container_name or set_env_vars is not None or remove_env_vars is not None or replace_env_vars is not None or remove_all_env_vars or cpu or memory or startup_command is not None or args is not None
+    update_map['registry'] = registry_server or registry_user or registry_pass
+
+    if tags:
+        _add_or_update_tags(new_containerappsjob, tags)
+
+    if workload_profile_name:
+        new_containerappsjob["properties"]["workloadProfileName"] = workload_profile_name
+
+        parsed_managed_env = parse_resource_id(containerappsjob_def["properties"]["environmentId"])
+        managed_env_name = parsed_managed_env['name']
+        managed_env_rg = parsed_managed_env['resource_group']
+        managed_env_info = None
+        try:
+            managed_env_info = ManagedEnvironmentClient.show(cmd=cmd, resource_group_name=managed_env_rg, name=managed_env_name)
+        except:
+            pass
+
+        if not managed_env_info:
+            raise ValidationError("Error parsing the managed environment '{}' from the specified containerappjob".format(managed_env_name))
+
+        ensure_workload_profile_supported(cmd, managed_env_name, managed_env_rg, workload_profile_name, managed_env_info)
+
+    # replicaConfiguration
+    if update_map["replicaConfigurations"]:
+        new_containerappsjob["properties"]["configuration"] = {} if "configuration" not in new_containerappsjob["properties"] else new_containerappsjob["properties"]["configuration"]
```

Comment:
consider safe_set

Reply:
will update logic to use safe_get in next PR